src/app/main/component/user/components/profile/profile-dashboard/profile-dashboard.component.ts

```diff
@@ -181,14 +206,21 @@ export class ProfileDashboardComponent implements OnInit, OnDestroy {
   }
 
   getUserEvents(): void {
```

Comment:
Since `getUserFavouriteEvents` and `getUserEvents` share similar functionality, we can refactor the shared logic into a a generalized `loadEvents` method to handle pagination and loading state:
`loadEvents(hasPage, page, isActive, List)`